### DISCUSSION TITLE
serde_as with generic types

### DISCUSSION THREAD

**Discussion by JosiahParry** [jonasbb/serde_with#663] [Q&A] [answered]
I'm trying to use serde_as with this example from the serde github issues. Where we have a custom deserializer and serializer for `[T; N]` https://github.com/serde-rs/serde/issues/1937#issuecomment-812137971. I have a struct 

```rust
pub struct Line<const N: usize> {
    #[serde_as(as = "Vec<ArrayVisitor>")]
    pub points: Vec<[f64; N]>
}
```

The ArrayVisitor: 
```rust
 struct ArrayVisitor<T, const N: usize>(PhantomData<T>);
```

When i try to compile I get the following issue: 

```
note: struct defined here, with 2 generic parameters: `T`, `N`
  --> src/de_array.rs:22:16
   |
22 |     pub struct ArrayVisitor<T, const N: usize>(PhantomData<T>);
   |                ^^^^^^^^^^^^ -  --------------
   = note: this error originates in the attribute macro `serde_as` (in Nightly builds, run with -Z macro-backtrace for more info)
help: add missing generic arguments
   |
22 | #[serde_as]<T, N>
   |            ++++++
```

How can I pass in the generic types or avoid this?

**Comment by jonasbb** [comment, marked as answer]
You can specify the generic types on the `ArrayVisitor` as usual. Something like this:

```rust
#[serde_as(as = "Vec<ArrayVisitor<f64, N>>")]
```

The correct values depend on your `SerializeAs` trait implementation, that you did not provide here.
